ember-primitives/src/components/menu.gts

```diff
@@ -0,0 +1,233 @@
+  Element: HTMLElement;
+  Args: {
+    Named: {
+      isOpen: Cell<boolean>;
+      triggerElement: Cell<HTMLElement>;
+    };
+  };
+}>((element, _: [], { isOpen, triggerElement }) => {
+  // focus first focusable element on the content
+  const tabster = getTabster(window);
+  const firstFocusable = tabster?.focusable.findFirst({
+    container: element,
+  });
+
+  firstFocusable?.focus();
+
+  // listen for "outside" clicks
+  function onDocumentClick(e: MouseEvent) {
+    if (
+      isOpen.current &&
+      e.target &&
+      !element.contains(e.target as HTMLElement) &&
+      !triggerElement.current?.contains(e.target as HTMLElement)
+    ) {
+      isOpen.current = false;
+    }
+  }
+
+  // listen for the escape key
+  function onDocumentKeydown(e: KeyboardEvent) {
+    if (isOpen.current && e.key === 'Escape') {
+      isOpen.current = false;
+    }
+  }
+
+  document.addEventListener('click', onDocumentClick);
+  document.addEventListener('keydown', onDocumentKeydown);
+
+  return () => {
+    document.removeEventListener('click', onDocumentClick);
+    document.removeEventListener('keydown', onDocumentKeydown);
+  };
+});
+
+const Content: TOC<{
+  Element: HTMLDivElement;
+  Args: {
+    triggerElement: Cell<HTMLElement>;
+    contentId: string;
+    isOpen: Cell<boolean>;
+    PopoverContent: PopoverSignature['Blocks']['default'][0]['Content'];
+  };
+  Blocks: { default: [{ Item: typeof Item; Separator: typeof Separator }] };
+}> = <template>
+  {{#if @isOpen.current}}
+    <@PopoverContent
+      id={{@contentId}}
+      role="menu"
+      data-tabster={{TABSTER_CONFIG_CONTENT}}
+      tabindex="0"
+      {{installContent isOpen=@isOpen triggerElement=@triggerElement}}
+      {{on "click" @isOpen.toggle}}
+      ...attributes
+    >
+      {{yield (hash Item=Item Separator=Separator)}}
+    </@PopoverContent>
+  {{/if}}
+</template>;
+
+const installTrigger = modifier<{
+  Element: HTMLElement;
+  Args: { Named: { triggerElement: Cell<HTMLElement> } };
+}>((element, _: [], { triggerElement }) => {
+  triggerElement.current = element;
+});
+
+const Trigger: TOC<{
+  Element: HTMLButtonElement;
+  Args: {
+    triggerElement: Cell<HTMLElement>;
+    contentId: string;
+    isOpen: Cell<boolean>;
+    hook: PopoverSignature['Blocks']['default'][0]['hook'];
+  };
+  Blocks: { default: [] };
+}> = <template>
+  <button
+    data-tabster={{TABSTER_CONFIG_TRIGGER}}
+    type="button"
+    aria-controls={{if @isOpen.current @contentId}}
+    aria-haspopup="menu"
+    aria-expanded={{if @isOpen.current "true" "false"}}
+    {{@hook}}
+    {{installTrigger triggerElement=@triggerElement}}
+    {{on "click" @isOpen.toggle}}
+    ...attributes
+  >
+    {{yield}}
+  </button>
+</template>;
+
+const IsOpen = () => cell<boolean>(false);
+const TriggerElement = () => cell<HTMLElement>();
+
+export const Menu: TOC<Signature> = <template>
+  {{#let (IsOpen) (uniqueId) (TriggerElement) as |isOpen contentId triggerEl|}}
```

Comment:
uniqueId doesn't exist pre 5.1, I think, and in order to support ember-source 4.12, maybe this `Menu` component should be a class component, and we use `guidFor(this)`?

Reply:
Done. 👍